libs/blocks/merch/threeInOne.js

```diff
@@ -0,0 +1,96 @@
+import { createTag, getConfig } from '../../utils/utils.js';
+import { MODAL_TYPE_3_IN_1 } from './constants.js';
+
+export const MSG_SUBTYPE = {
+  AppLoaded: 'AppLoaded',
+  EXTERNAL: 'EXTERNAL',
+  SWITCH: 'SWITCH',
+  RETURN_BACK: 'RETURN_BACK',
+  OrderComplete: 'OrderComplete',
+  Error: 'Error',
+  Close: 'Close',
+};
+
+export const LANA_OPTIONS = {
+  clientId: 'merch-at-scale',
+  sampleRate: 10,
+  tags: 'three-in-one',
+};
+
+export const handle3in1IFrameEvents = ({ data: msgData }) => {
+  let parsedMsg = null;
+  try {
+    parsedMsg = JSON.parse(msgData);
+  } catch (error) {
+    return;
+  }
+  const { app, subType, data } = parsedMsg || {};
+  if (app !== 'ucv3') return;
+  window.lana?.log(`3-in-1 modal: ${subType}`, LANA_OPTIONS);
+  switch (subType) {
+    case MSG_SUBTYPE.AppLoaded:
+      document.querySelector('.three-in-one iframe')?.classList?.remove('loading');
+      document.querySelector('.three-in-one sp-theme')?.remove();
+      break;
+    case MSG_SUBTYPE.EXTERNAL:
```

Comment:
I see the same action being executed for `MSG_SUBTYPE.EXTERNAL`, `MSG_SUBTYPE.SWITCH`, `MSG_SUBTYPE.RETURN_BACK`. Maybe the code can be refactored a bit here.